packages/webcrypto-ed25519-polyfill/src/__tests__/index-test.ts

```diff
@@ -0,0 +1,185 @@
+            expect.assertions(1);
+            const mockKeyPair = {};
+            (generateKeyPolyfill as jest.Mock).mockReturnValue(mockKeyPair);
+            const keyPair = await globalThis.crypto.subtle.generateKey('Ed25519', /* extractable */ false, [
+                'sign',
+                'verify',
+            ]);
+            expect(keyPair).toBe(mockKeyPair);
+        });
+    });
+    describe('when required in an environment that supports Ed25519', () => {
+        beforeEach(() => {
+            jest.isolateModules(() => {
+                require('../index');
+            });
+        });
+        it('overrides `generateKey`', () => {
+            expect(globalThis.crypto.subtle.generateKey).not.toBe(originalGenerateKey);
+        });
+        it.each([
+            { __variant: 'P256', name: 'ECDSA', namedCurve: 'P-256' },
+            { __variant: 'P384', name: 'ECDSA', namedCurve: 'P-384' } as EcKeyGenParams,
+            { __variant: 'P521', name: 'ECDSA', namedCurve: 'P-521' } as EcKeyGenParams,
+            ...['RSASSA-PKCS1-v1_5', 'RSA-PSS'].flatMap(rsaAlgoName =>
+                ['SHA-1', 'SHA-256', 'SHA-384', 'SHA-512'].map(
+                    hashName =>
+                        ({
+                            __variant: hashName,
+                            hash: { name: hashName },
+                            modulusLength: 2048,
+                            name: rsaAlgoName,
+                            publicExponent: new Uint8Array([0x01, 0x00, 0x01]),
+                        } as RsaHashedKeyGenParams)
+                )
+            ),
+        ])('calls the original `generateKey` when the algorithm is $name/$__variant', async algorithm => {
+            expect.assertions(1);
+            await globalThis.crypto.subtle.generateKey(algorithm, /* extractable */ false, ['sign', 'verify']);
+            expect(originalGenerateKey).toHaveBeenCalled();
+        });
+        it('delegates the call to the original `generateKey` when the algorithm is "Ed25519"', async () => {
+            expect.assertions(1);
+            const mockKeyPair = {};
+            (originalGenerateKey as jest.Mock).mockResolvedValue(mockKeyPair);
+            await expect(
+                globalThis.crypto.subtle.generateKey('Ed25519', /* extractable */ false, ['sign', 'verify'])
+            ).resolves.toBe(mockKeyPair);
+        });
+        it('calls the original `generateKey` once per call to `generateKey` when the algorithm is "Ed25519" but never again (parallel version)', async () => {
+            expect.assertions(1);
+            await Promise.all([
+                globalThis.crypto.subtle.generateKey('Ed25519', /* extractable */ false, ['sign', 'verify']),
+                globalThis.crypto.subtle.generateKey('Ed25519', /* extractable */ false, ['sign', 'verify']),
+            ]);
+            expect(originalGenerateKey).toHaveBeenCalledTimes(2);
```

Comment:
Is the "but never again" supposed to be in the name of this test? It looks like this will call the original function once per call, always? (Same question for the serial version below)

Reply:
![giphy](https://github.com/solana-labs/solana-web3.js/assets/13243/4bf932d1-8a34-47be-9d28-3183d4d8a6d4)